### DISCUSSION TITLE
Food sanity check and rebalance

### DISCUSSION THREAD

**Discussion by AlecWhite** [cataclysmbn/Cataclysm-BN#3406] [General]
Foodstuff needs a review and rebalance after all these years. 
A lot of food and dishes have very variable calorie output considering the materials it takes and the volume they have while others have lower calories than it should. 
Then we have some odd things with food and how many charges it should have. As it affects the total calories it have.

So I propose a food sanity check to rebalance the whole of comestibles as well as a debate to add to guidelines what kind of food system do we want for BN.
Should food be gamefy(balanced around gameplay)?
Should food be realistic(balanced around the [US Department of Agriculture food database](https://fdc.nal.usda.gov/))?
Or a mix of both?

Here is a list of the most calorie dense food per charge made by scarf.
You can see some inconsistencies(why fat is in the list but cracklings aren't? cracklings are divided by charges, a lot of them) and in general the most calorie dense single serving items.
```
┌───────┬─────────────────────┬──────────────────────────┬──────────┬─────────┬───────┐
│ (idx) │ id                  │ name                     │ calories │ charges │ total │
├───────┼─────────────────────┼──────────────────────────┼──────────┼─────────┼───────┤
│     0 │ "debug_nutrition"   │ "holy SPAM of debugging" │     2100 │       1 │  2100 │
│     1 │ "fat"               │ "chunk of fat"           │     1923 │       1 │  1923 │
│     2 │ "testcomest"        │ "diet pill"              │     1000 │       1 │  1000 │
│     3 │ "tallow"            │ "tallow"                 │      962 │       2 │  1924 │
│     4 │ "catfood"           │ "cat food"               │      800 │       1 │   800 │
│     5 │ "haggis"            │ "haggis"                 │      794 │       1 │   794 │
│     6 │ "fish_fried"        │ "batter fried fish"      │      748 │       1 │   748 │
│     7 │ "barley"            │ "barley"                 │      743 │       4 │  2972 │
│     8 │ "sandwich_pbm"      │ "PB&M sandwich"          │      712 │       1 │   712 │
│     9 │ "porridge_cooked"   │ "boiled grain"           │      705 │       1 │   705 │
│    10 │ "fish_bagel"        │ "fish and spinach bagel" │      700 │       1 │   700 │
│    11 │ "mycus_fruit"       │ "mycus fruit"            │      694 │       1 │   694 │
│    12 │ "wheat"             │ "wheat"                  │      667 │       1 │   667 │
│    13 │ "sandwich_pb"       │ "peanut butter sandwich" │      658 │       1 │   658 │
│    14 │ "pizza_supreme"     │ "supreme pizza"          │      641 │       1 │   641 │
│    15 │ "sandwich_fairy"    │ "fairy bread"            │      640 │       1 │   640 │
│    16 │ "onion_rings"       │ "onion ring"             │      639 │       1 │   639 │
│    17 │ "royal_beef"        │ "royal beef"             │      637 │       1 │   637 │
│    18 │ "starch"            │ "starch"                 │      620 │       1 │   620 │
│    19 │ "buckwheat"         │ "buckwheat"              │      616 │       2 │  1232 │
│    20 │ "can_coconut"       │ "coconut milk"           │      573 │       2 │  1146 │
│    21 │ "pizza_meat"        │ "meat pizza"             │      566 │       4 │  2264 │
│    22 │ "glazed_tenderloin" │ "glazed tenderloins"     │      560 │       3 │  1680 │
│    23 │ "coffee_bean"       │ "coffee beans"           │      549 │       1 │   549 │
│    24 │ "honey_ant"         │ "honey ball"             │      541 │       2 │  1082 │
│    25 │ "sweet_sausage"     │ "sweet sausage"          │      539 │       1 │   539 │
│    26 │ "sandwich_honey"    │ "honey sandwich"         │      535 │       1 │   535 │
│    27 │ "tofu_stirfry"      │ "tofu stirfry"           │      525 │       4 │  2100 │
│    28 │ "sandwich_cucumber" │ "cucumber sandwich"      │      516 │       1 │   516 │
│    29 │ "sandwich_veggy"    │ "vegetable sandwich"     │      502 │       1 │   502 │
└───────┴─────────────────────┴──────────────────────────┴──────────┴─────────┴───────┘
```

**Comment by scarf005** [comment]
> Here is a list of the most calorie dense food per charge made by scarf.

![](https://user-images.githubusercontent.com/54838975/273486731-ff3f1082-f0ca-4c09-a75c-0053a3776d06.png)

for reference, [scripts can be ran like this](https://github.com/scarf005/catjazz?tab=readme-ov-file#user-usage-example) (you need to have `deno` installed in your PC)
```sh
deno run --allow-read https://deno.land/x/catjazz@v0.0.2/examples/calories.ts --path ./cataclysm/data/json/items
```

**Comment by scarf005** [comment]
about the food balance matter, i'm more leaned to gamifying it, as currently most foods are abundant and their characteristics are almost cosmetic. 

it should be kind of its own discussion, but i think this could be achieved by making `material` system more complex, e.g sandwich of `material { wheat: 2, meat: 3, vegetable: 1, dairy: 1 }`

**Comment by Relgar** [comment]
Food in many games usually serve single purposes - whether to heal, to buff, or stave off starvation. I think that BN is deep enough to go for all 3 to be honest. I suggest a few ways to tweak our current situation.

1. Reduce food drop rates in general - an innawoods character appears to be living in a tropical paradise the moment they hit a forest. They can pull as much food as they can carry from the multitude of bushes, fruit trees, shrubs, etc. A chance for gaining nothing could be implemented in those cases. This is the easiest solution but doesn't really address the design issues with food abundance.

2. Make base food items provide less nutrition - some games reward you for the action of cooking, and I think this is a great idea to implement here. For example, if you have an apple, it might provide xyz calories, but if it's baked into an apple pie, have it add all caloric values of apples and other ingredients together and give it a multiplication bonus of 10-20% more nutrition. This way, we encourage players to develop their cooking and seek out new tools to improve the output of their food. Higher difficulty food takes more time and provides a larger tier bonus. This means that a dish that is composed of multiple cooked items is most economical. We could also tier it by having different methods of cooking give different multiplier bonuses, or have them tie to the buff idea below.

3. Make special food items provide buffs - some foods may not necessarily provide a full stomach, but may have other effects like increasing stats / reducing damage intake / increase damage output / making you learn faster. These effects can be high-intensity and short duration, or low-intensity and long duration. Happiness is a current example of this mechanic - some foods make you super happy. This may have some crossover with the chemistry or drug type items. Think of the vitamin system in DDA but instead of making life terrible by not taking the required amount of vitamins, give the player buffs for taking care of their character.

**Comment by ghost** [comment]
Use the FDA's website when making changes to raw ingredients and ensure food properly inherits those calories and nutrients. https://fdc.nal.usda.gov/fdc-app.html#/food-details/169097/nutrients
